pkg/controller/stacks/install/installjob.go

```diff
@@ -213,29 +223,41 @@ func (jc *stackInstallJobCompleter) readPodLogs(namespace, name string) (*bytes.
 // Expected resources are CRD, Stack, & StackDefinition
 // nolint:gocyclo
 func (jc *stackInstallJobCompleter) createJobOutputObject(ctx context.Context, obj *unstructured.Unstructured,
-	i stacks.KindlyIdentifier, job *batchv1.Job) error {
+	i v1alpha1.StackInstaller, job *batchv1.Job) error {
 
 	// if we decoded a non-nil unstructured object, try to create it now
 	if obj == nil {
 		return nil
 	}
 
-	// when the current object is a Stack object, make sure the name and namespace are
-	// set to match the current StackInstall (if they haven't already been set). Also,
-	// set the owner reference of the Stack to be the StackInstall.
-	if isStackObject(obj) || isStackDefinitionObject(obj) {
+	// Modify Stack and StackDefinition resources based on StackInstall
+	isStack := isStackObject(obj)
+	isStackDefinition := !isStack && isStackDefinitionObject(obj)
```

Comment:
Do we need the `!isStack &&`? Can both be true?

Reply:
The unstructured `obj` can only have one `apiVersion` + `Kind` combination.  These functions are exclusive.

Reply:
I'd like to eventually clean this up by removing the Stack vs StackDefinition vs CRD guesswork.

https://github.com/crossplane/crossplane/compare/master...displague:unpack-with-structure?expand=1 (WIP)